portal/components/react-admin/base/resources/candidateProfile/candidate-profile.js

```diff
@@ -0,0 +1,204 @@
+import React from "react";
+import {
+  List,
+  Datagrid,
+  Pagination,
+  FunctionField,
+  TopToolbar,
+  sanitizeListRestProps,
+  Filter,
+  SearchInput,
+  ExportButton,
+  downloadCSV,
+} from "react-admin";
+import { makeStyles, Typography } from "@material-ui/core";
+import jsonExport from "jsonexport/dist";
+
+const SearchFilter = (props) => {
+  console.log("Props:", props);
+  return (
+    <Filter {...props}>
+      <SearchInput
+        placeholder="Search by Name *"
+        source="name"
+        className="searchBar"
+        alwaysOn
+      />
+    </Filter>
+  );
+};
+const exporter = (records) => {
+  const recordsForExport = records.map((record) => {
+    let age = getAge({
+      start: record.DOB,
+      end: null,
+    }).years;
+
+    return {
+      "Candidate Name": record.name ? record.name : "NONE",
+      "Mobile Number": record.mobile_number ? record.mobile_number : "NONE",
+      "Whatsapp Number": record.whatsapp_mobile_number
+        ? record.whatsapp_mobile_number
+        : "NONE",
+      District: record.district_name.name ? record.district_name.name : "NONE",
```

Comment:
Can we use optional chaining whenever a nested attribute is referenced?
`record.district_name?.name`, `record.highest_level_qualification?.highest_level_qualification_name` etc. - everywhere?